doc/amazon_sagemaker_debugger.rst

```diff
@@ -0,0 +1,356 @@
+        rule_parameters={
+          "threshold": "20.0" # this will be used to initialize 'threshold' param in your rule constructor
+        }
+    )
+
+    estimator = TensorFlow(
+        role=role
+        train_instance_count=1,
+        train_instance_type=train_instance_type,
+        rules=[
+            custom_gradient_rule
+        ]
+    )
+
+While initializing the custom rule through ``Rules.custom()``, for the value of ``source`` you can optionally specify a valid S3 location as well.
+
+
+Capture real-time TensorBoard data from the debugging hook
+==========================================================
+
+In addition to emitting and storing the debugging data useful for analyses, the debugging hook is also capable of emitting `TensorBoard <https://www.tensorflow.org/tensorboard>`__ data for you to point your TensorBoard application at and visualize.
+
+To enable the debugging hook to emit TensorBoard data, you'd need to specify the new option ``TensorBoardOutputConfig`` as:
+
+.. code:: python
+
+    from sagemaker.debugger import TensorBoardOutputConfig
+
+    tensorboard_output_config = TensorBoardOutputConfig(
+        s3_output_path='s3://path/for/tensorboard/data/emission',
+        container_local_output_path='/local/path/for/tensorboard/data/emission'
+    )
+
+    estimator = TensorFlow(
+        role=role
+        train_instance_count=1,
+        train_instance_type=train_instance_type,
+        tensorboard_output_config=tensorboard_output_config
+    )
+
+When you call ``fit()`` on the ``estimator``, the debugging hook will upload the generated TensorBoard data in near real-time to an S3 path derived from the value of ``s3_output_path`` provided in the configuration:
+
+.. code::
+
+    s3://{destination-bucket-prefix}/tensorboard-output/
+
+To access the S3 path where the tensorboard data is stored, you can do:
+
+.. code:: python
+
+    tensorboard_s3_output_path = estimator.latest_job_tensorboard_artifacts_path()
+
+
+Interactive analysis using SageMaker Debugger SDK and visualizations
+====================================================================
+
+`Amazon SageMaker Debugger SDK <https://github.com/awslabs/sagemaker-debugger>`__ also allows you to do interactive analyses on the debugging data produced from the training job run and render visualizations off it. After calling ``fit()`` on the estimator, you can use the Debugger SDK to load the saved data in a SageMaker Debugger ``trial`` and do analyses on the data:
+
+.. code:: python
+
+    from smdebug.trials import create_trial
+
+    s3_output_path = estimator.latest_job_debugger_artifacts_path()
+    trial = create_trial(s3_output_path)
+
+
+For a tutorial on what all you can do after creating the trial and how to visualize the results, see `SageMaker Debugger - Visualizing Debugging Results <https://github.com/awslabs/amazon-sagemaker-examples/blob/master/sagemaker-debugger/mnist_tensor_plot/mnist-tensor-plot.ipynb>`__.
+
```

Comment:
link to sagemaker-debugger analysis page.

Reply:
For a tutorial on what all you -> For a tutorial on what you

Reply:
* [@Vikas-kum] Done.
* [@aws-rbs] Fixed.